kernel/derive-no-peripheral-management/README.md

```diff
@@ -0,0 +1,63 @@
+No Peripheral Management
+========================
+
+Tock generally provides tooling to support managed control of peripherals.
+However, some peripherals may not require any support. These peripherals
+can use this derive to handle all the boilerplate automatically.
+
+For peripherals that require no runtime management, Tock still prefers the use
+of a PeripheralManager, as it reduces the `unsafe` surface and presents an easy
+mechanism for possible future enhancements that may wish to change an unmanaged
+peripheral into a managed one.
+
+Example / Usage
+---------------
+
+ - Update your Cargo.toml:
+
+    derive-no-peripheral-managemenet = { path = "../../kernel/derive-no-peripheral-managemenet" }
+
+ - Add this crate to your local lib.rs:
+
+    #[macro_use]
+    extern crate derive_no_clock_control;
+
+ - Then your peripheral should look something like this:
+
+```rust
+use kernel::StaticRef;
+use kernel::common::VolatileCell;
+use kernel::common::peripherals::PeripheralManager;
+
+/// The MMIO Structure.
+#[repr(C)]
+#[allow(dead_code)]
+pub struct TestRegisters {
+    control: VolatileCell<u32>,
+    interrupt_mask: VolatileCell<u32>,
+}
+
+/// The Tock object that holds all information for this peripheral.
+#[derive(NoPeripheralManagement)]
```

Comment:
Along the same thread as Brad and Amit's comments, I'm unsure what the purpose of deriving NoPeripheralManagement is. It's super neat that it's possible. However, it feels like in practice things should add peripheral management the normal way and just leave some of the functions as no-ops. What's the advantage of this over that?

Reply:
This is _exactly_ that, only you don't have to write the empty no-op functions yourself, the derive writes them for you.